File: Comparación entre complidores e Interpretes/c/fib_iter.c
/* Fibonacci iterativo.
   Complejidad: tiempo O(n), memoria O(1). */
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

static unsigned long long fib_iter(int n) {
    unsigned long long a = 0, b = 1, t;
    for (int i = 0; i < n; ++i) { t = b; b = a + b; a = t; }
    return a;
}

int main(int argc, char** argv) {
    if (argc < 2) { fprintf(stderr, "uso: fib_iter N\n"); return 1; }
    int n = atoi(argv[1]);
    printf("%llu\n", fib_iter(n));
    return 0;
}

File: Comparación entre complidores e Interpretes/py/fib_iter.py
# Fibonacci iterativo: O(n) tiempo, O(1) memoria.
import sys

def fib_iter(n: int) -> int:
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("uso: python fib_iter.py N"); raise SystemExit(1)
    n = int(sys.argv[1])
    print(fib_iter(n))
